bibtex-actions.el

```diff
@@ -67,19 +67,19 @@
   (bibtex-actions-read-backend))
 
 (defun bibtex-actions--completing-read ()
-  "Read bibtex-completion entries for completion using 'completing-read'."
+  "Read bibtex-completion entries for completion using 'completing-read-multiple'."
   (bibtex-completion-init)
   (when-let ((candidates (bibtex-actions--get-candidates))
              (chosen
-              (completing-read
+              (completing-read-multiple
                "BibTeX entries: "
                (lambda (string predicate action)
                  (if (eq action 'metadata)
                      '(metadata
-                       ;; (annotation-function . bibtex-completion--annotation)
+                       ;; TODO (annotation-function . bibtex-completion--annotation)
                        (category . bibtex))
                    (complete-with-action action candidates string predicate))))))
-    (cdr (assoc chosen candidates))))
+    (mapcar (lambda (choice) (cdr (assoc choice candidates))) chosen)))
```

Comment:
So with changes last night, if I understand right, the read function is getting fed nil results?

So, `bibtex-actions-insert-key` results in a comma-separated list of empty results, and some of the other insert commands complain about wrong type.

And here's what the read function returns:

``` elisp
ELISP> (bibtex-actions--completing-read)
(nil nil nil nil nil)
```

That makes me think there's something wrong with this line, but I can't see any problem.

@oantolin?

Reply:
I think the problem is simply that you are using the default action of `crm-separator`, which is a comma (well, a regexp that matches optional spaces, then a comma, then optional spaces), while most formatted bibtex entries also contain a comma, so `completing-read-multiple` gets confused. You should pick a separator that is unlikely to occur in the formatted bibtex entries. I noticed the bibtex file I was testing with had no ampersands so I set `crm-separator` to `"\\s-*&\\s-*"` and it seemed to work then.

Reply:
Ah, that makes sense!

I believe in BibTeX you have to escape ampersands with a backslash.

How do I set the `crm-separator`?

Reply:
Ah, think I figured it out.

Only thing is, I have to manually add the separator.